mantle/cmd/kola/testiso.go

```diff
@@ -233,22 +271,62 @@ RequiredBy=coreos-installer.target
 RequiredBy=multi-user.target`, nmConnectionId, nmConnectionFile)
 
 func init() {
+	archPatterns = getPatterns()
 	cmdTestIso.Flags().BoolVarP(&instInsecure, "inst-insecure", "S", false, "Do not verify signature on metal image")
-	cmdTestIso.Flags().BoolVarP(&nopxe, "no-pxe", "P", false, "Skip testing live installer PXE")
-	cmdTestIso.Flags().BoolVarP(&noiso, "no-iso", "", false, "Skip testing live installer ISO")
 	cmdTestIso.Flags().BoolVar(&console, "console", false, "Connect qemu console to terminal, turn off automatic initramfs failure checking")
 	cmdTestIso.Flags().BoolVar(&pxeAppendRootfs, "pxe-append-rootfs", false, "Append rootfs to PXE initrd instead of fetching at runtime")
 	cmdTestIso.Flags().StringSliceVar(&pxeKernelArgs, "pxe-kargs", nil, "Additional kernel arguments for PXE")
-	cmdTestIso.Flags().BoolVar(&addNmKeyfile, "add-nm-keyfile", false, "Add NetworkManager connection keyfile")
-	cmdTestIso.Flags().StringSliceVar(&scenarios, "scenarios", []string{scenarioPXEInstall, scenarioISOOfflineInstall, scenarioPXEOfflineInstall, scenarioISOLiveLogin, scenarioISOAsDisk, scenarioMinISOInstall, scenarioMinISOInstallNm}, fmt.Sprintf("Test scenarios (also available: %v)", []string{scenarioISOInstall}))
+	cmdTestIso.Flags().StringSliceVar(&patterns, "patterns", archPatterns, "Test patterns available")
 	cmdTestIso.Args = cobra.ExactArgs(0)
 
 	root.AddCommand(cmdTestIso)
 }
 
+func checkIso(live string) error {
+
+	if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil && live  == "iso" {
+		return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
+	}
+	if kola.CosaBuild.Meta.BuildArtifacts.LiveKernel == nil && live == "kernel" {
+		return fmt.Errorf("build %s has no live installer kernel", kola.CosaBuild.Meta.Name)
+	}
+	return nil
+}
+
+func getPatterns() []string {
+	pltfrm = coreosarch.CurrentRpmArch()
+	var pattern []string
+	switch pltfrm {
+		case "x86_64":
+			pattern = tests_x86_64
+		case "ppc64le":
+			pattern = tests_ppc64le
+		case "s390x":
+			pattern = tests_s390x
+		case "aarch64":
+			pattern = tests_aarch64
+	}
+	return pattern
+}
+func matchPattern(pattern string, denyList []string) bool {
```

Comment:
Why not use the same `matchesPatterns` from `harness.go` instead? Notably, it uses `filepath.Match` instead of regex, which I think we want to match here so the semantics are consistent. That should fix the issue you've been having with `*` I think.

Reply:
The issue with `*` is a yaml  validation issue, it understands `*` as an alias.  We can by pass it adding quotation marks `pattern: "*.4k"`. However, we would need to allow quotation marks. Another issue I'm not sure yet, is if we allow `*` at the begging if we would need to add this support for kola run too

Reply:
For filepath.Match, the issue is:
pattern: miniso-install.uefi.*.4k
  - miniso-install.uefi.nm.4k:  ran
  - miniso-install.uefi.nm: Didn't run 
  -  miniso-install.uefi: Didn't run 

Using regex:
 pattern: miniso-install.uefi.*.4k
  - miniso-install.uefi.nm.4k: Didn't run
  - miniso-install.uefi.nm: Ran
  -  miniso-install.uefi:  Ran

I think what we want is the regex behavior

Reply:
> For filepath.Match, the issue is:
> pattern: miniso-install.uefi.*.4k
>  - miniso-install.uefi.nm.4k:  ran
>  - miniso-install.uefi.nm: Didn't run 
>  - miniso-install.uefi: Didn't run 

Hmm, that's weird. In a quick test locally:

```
$ cat main.go
package main

import (
        "fmt"
        "path/filepath"
)

func main() {
        tests := []string{
                "miniso-install.uefi.nm.4k",
                "miniso-install.uefi.nm",
                "miniso-install.uefi",
        }
        for _, test := range tests {
                matches, _ := filepath.Match("miniso-install.uefi.*.4k", test)
                fmt.Printf("%s: %v\n", test, matches)
        }
}
$ go build main.go
$ ./main
miniso-install.uefi.nm.4k: true
miniso-install.uefi.nm: false
miniso-install.uefi: false
```

Reply:
But BTW, based on https://github.com/coreos/coreos-assembler/pull/3298#discussion_r1061833545, I think we want to reorder the fields so that the firmware is last? That'll make pattern matching easier I thin.

Reply:
Found the issue with `filepath.Match`, thanks for checking it!